app/server/models/credit_transfer.py

```diff
@@ -172,19 +174,67 @@ def resolve_as_completed(self, existing_blockchain_txn=None):
             if self.recipient_user and self.recipient_user.transfer_card:
                 self.recipient_user.transfer_card.update_transfer_card()
 
-        if self.transfer_type == TransferTypeEnum.DEPOSIT and self.fiat_ramp:
+        if self.fiat_ramp and self.transfer_type in [TransferTypeEnum.DEPOSIT, TransferTypeEnum.WITHDRAWAL]:
             self.fiat_ramp.resolve_as_completed()
 
         if not existing_blockchain_txn:
             self.send_blockchain_payload_to_worker()
 
     def resolve_as_rejected(self, message=None):
+        if self.fiat_ramp and self.transfer_type in [TransferTypeEnum.DEPOSIT, TransferTypeEnum.WITHDRAWAL]:
+            self.fiat_ramp.resolve_as_rejected()
+
         self.resolved_date = datetime.datetime.utcnow()
         self.transfer_status = TransferStatusEnum.REJECTED
 
         if message:
             self.resolution_message = message
 
+    def check_sender_txns_limits(self):
+        if self.sender_user is None:
+            # skip if an exchange
+            return
+
+        relevant_txn_limits = [txn for txn in self.sender_user.get_txn_limits(credit_transfer=self) if str(self.transfer_type) in txn.get('txn_type')]
```

Comment:
`limit for limit in...`?

Reply:
there's naming ambiguity between `limits` and `limit_rules`, and that's my fault lol

Reply:
I'd consider requiring that `txn_type` (or `transfer_type`) be a list such as `applied_to_transfer_types` so we don't get caught be cases such as `'FUND' in 'REFUND'` (returns true).

Reply:
Also `if str(self.transfer_type) in txn.get('txn_type')` looks more like an application filter than part of the limit rule itself. perhaps this should be refactored into the `get_txn_limits` method? Equally, in hindsight `get_txn_limits` probably makes more sense as a method on the transfer itself, so that we wouldn't have to redefine it for orgs or other transfer acount types in the future